python/glow/hail/tests/test_from_matrix_table.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2019 The Glow Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from glow.hail import functions
+import hail as hl
+from pyspark.sql import functions as fx
+from pyspark.sql import Row
+from pyspark.sql.types import ArrayType, StructType
+import pytest
+
+
+# Check that structs have the same fields and datatypes (not necessarily metadata), in any order
+def __compare_struct_types(s1, s2, ignore_fields=[]):
+    s1_fields = [f for f in s1.fields if f.name not in ignore_fields]
+    s2_fields = [f for f in s2.fields if f.name not in ignore_fields]
+    assert set([f.name for f in s1_fields]) == set([f.name for f in s2_fields])
+    for f1 in s1_fields:
+        matching_fields = [f2 for f2 in s2_fields if f1.name == f2.name]
+        assert (len(matching_fields) == 1)
+        m = matching_fields[0]
+        if isinstance(m.dataType, ArrayType) and isinstance(m.dataType.elementType, StructType):
+            __compare_struct_types(f1.dataType.elementType, m.dataType.elementType, ignore_fields)
+        else:
+            assert f1.dataType == m.dataType
+
+
+def __compare_round_trip(spark,
```

Comment:
nit: This isn't really a round trip, right? Perhaps it would be better to say `assert_adapter_matches_reader` or `assert_adapter_matches_glow_reader`